src/eos-updater-poll.c

```diff
@@ -257,6 +257,197 @@ async_result_cb (GObject      *source_object,
+  gboolean is_update = FALSE;
+  g_autofree gchar *checksum = NULL;
+  g_autoptr(GVariant) commit = NULL;
+  g_autoptr(OstreeSysroot) sysroot = ostree_sysroot_new_default ();
+  g_autoptr(OstreeDeployment) booted_deployment = NULL;
+
+  g_return_val_if_fail (out_is_update != NULL, FALSE);
+  g_return_val_if_fail (out_update_ref_info != NULL, FALSE);
+  g_return_val_if_fail (error == NULL || *error == NULL, FALSE);
+
+  if (!ostree_sysroot_load (sysroot, NULL, error))
+    return FALSE;
+
+  booted_deployment = eos_updater_get_booted_deployment_from_loaded_sysroot (sysroot,
+                                                                             error);
+
+  if (booted_deployment == NULL)
+    return FALSE;
+
+  if (!get_booted_refspec (booted_deployment,
+                           &booted_refspec,
+                           &remote,
+                           &ref,
+                           &collection_ref, error))
+    return FALSE;
+
+  if (!fetch_latest_commit (repo,
+                            cancellable,
+                            booted_refspec,
+                            NULL,
+                            &checksum,
+                            &new_refspec,
+                            &version,
+                            error))
+    return FALSE;
+
+  if (!is_checksum_an_update (repo, checksum, &commit, error))
+    return FALSE;
+
+  is_update = (commit != NULL);
+  *out_is_update = is_update;
+
+  if (is_update)
+    {
+      out_update_ref_info->refspec = g_steal_pointer (&booted_refspec);
+      out_update_ref_info->remote = g_steal_pointer (&remote);
+      out_update_ref_info->ref = g_steal_pointer (&ref);
+      out_update_ref_info->collection_ref = g_steal_pointer (&collection_ref);
+      out_update_ref_info->new_refspec = g_steal_pointer (&new_refspec);
+      out_update_ref_info->checksum = g_steal_pointer (&checksum);
+      out_update_ref_info->version = g_steal_pointer (&version);
+      out_update_ref_info->commit = g_steal_pointer (&commit);
+    }
+  else
+    {
+      update_ref_info_clear (out_update_ref_info);
+    }
+
+  return TRUE;
+}
+
+static gboolean
+check_for_update_following_checkpoint_commits (OstreeRepo     *repo,
+                                               UpdateRefInfo  *out_update_ref_info,
+                                               GCancellable   *cancellable,
+                                               GError        **error)
+{
+  g_autofree gchar *upgrade_refspec = NULL;
+  g_autofree gchar *remote = NULL;
+  g_autofree gchar *ref = NULL;
+  g_autoptr(OstreeCollectionRef) collection_ref = NULL;
+  g_autofree gchar *new_refspec = NULL;
+  g_autofree gchar *version = NULL;
+  g_autofree gchar *checksum = NULL;
+  g_autoptr(GVariant) commit = NULL;
+
+  g_return_val_if_fail (out_update_ref_info != NULL, FALSE);
+  g_return_val_if_fail (error == NULL || *error == NULL, FALSE);
+
+  if (!get_refspec_to_upgrade_on (&upgrade_refspec, NULL, NULL, NULL, error))
```

Comment:
@smspillaz why doesn't this fill in `remote`, `ref` and `collection_ref`? It seems like it will just end up hitting the `update_ref_info.collection_ref == NULL` check you added in `metadata_fetch_new`. Am I missing something?

Reply:
Hmm, I see your point. The fact that the tests still work is ... concerning. I'll quickly look into what the tests are doing.

Reply:
Okay, so it looks like the "new" metadata fetch code fails unconditionally (whoops) and we fall back to the old code, which is the reason why all the tests still pass.

I'll push a branch to fix this but I do wonder if we should set `EOS_UPDATER_DISABLE_FALLBACK_FETCHERS` in the tests to catch this in the future?

Reply:
Fixed in #198. See my comments on https://github.com/endlessm/eos-updater/pull/198#pullrequestreview-116846421 about `EOS_UPDATER_DISABLE_FALLBACK_FETCHERS`.